### PR TITLE
Make KubeJobNotCompleted configurable in prometheus operator config

### PR DESCRIPTION
**Pull request by eg-ayoub**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 ## Release 125.0.6 (In development)
 
+### Additions
```
Comment (TeddyAndrieux) [inline CHANGELOG.md]: To me it's more an `Enhancements`

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
...
+
+- Add alert configuration for KubeJobNotCompleted alert
+  (PR[4128](https://github.com/scality/metalk8s/pull/4128))
```
Comment (TeddyAndrieux) [inline CHANGELOG.md]: I would reword it a bit like "Make `KubeJobNotCompleted` alert time configurable"

File: salt/metalk8s/addons/prometheus-operator/deployed/kube-alerts-rules.sls [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    )
+%}
+{%- set prometheus = salt.metalk8s_service_configuration.get_service_conf(
+        'metalk8s-monitoring', 'metalk8s-prometheus-config', prometheus_defaults
+    )
+%}
+{%- set rules = prometheus.get('spec', {}).get('rules', {}) %}
+
+{%- raw %}
+spec:
```
Comment (TeddyAndrieux) [inline salt/metalk8s/addons/prometheus-operator/deployed/kube-alerts-rules.sls:24]: You miss all the object metadata here + apiversion & co

File: salt/metalk8s/addons/prometheus-operator/deployed/kube-alerts-rules.sls [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  groups:
+  - name: kubernetes-apps
+    rules:
+    - alert: KubeJobNotCompleted
+      annotations:
+        description: Job {{ $labels.namespace }}/{{ $labels.job_name }} is taking
+          more than {% endraw %} rules.kube_apps.kube_job_not_completed.warning.hours {% raw %} hours to complete.
```
Comment (TeddyAndrieux) [inline salt/metalk8s/addons/prometheus-operator/deployed/kube-alerts-rules.sls]: The jinja variable will not be replaced here to me

```suggestion
          more than {% endraw %}{{ rules.kube_apps.kube_job_not_completed.warning.hours }}{% raw %} hours to complete.
```

File: salt/metalk8s/addons/prometheus-operator/deployed/kube-alerts-rules.sls [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+        runbook_url: https://runbooks.prometheus-operator.dev/runbooks/kubernetes/kubejobnotcompleted
+        summary: Job did not complete in time
+      expr: |-
+        time() - max by(namespace, job_name, cluster) (kube_job_status_start_time{job="kube-state-metrics", namespace=~".*"}
+          and
+        kube_job_status_active{job="kube-state-metrics", namespace=~".*"} > 0) > {% endraw %} rules.kube_apps.kube_job_not_completed.warning.hours {% raw %}*60*60
```
Comment (TeddyAndrieux) [inline salt/metalk8s/addons/prometheus-operator/deployed/kube-alerts-rules.sls]: ditto

**Review by TeddyAndrieux — CHANGES_REQUESTED** [review]

File: salt/metalk8s/addons/prometheus-operator/deployed/kube-alerts-rules.sls [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    app.kubernetes.io/managed-by: salt
+    app.kubernetes.io/name: prometheus-operator
+    app.kubernetes.io/part-of: metalk8s
+    app.kubernetes.io/version: 45.5.0
+    chart: kube-prometheus-stack-45.5.0
+    heritage: metalk8s
+    metalk8s.scality.com/monitor: ''
+    release: prometheus-operator
+  name: prometheus-operator-kubernetes-apps
+  namespace: metalk8s-monitoring
```
Comment (eg-ayoub) [inline salt/metalk8s/addons/prometheus-operator/deployed/kube-alerts-rules.sls]: fetched these from the chart. @TeddyAndrieux do any of them have to go ?

Comment (TeddyAndrieux) [reply salt/metalk8s/addons/prometheus-operator/deployed/kube-alerts-rules.sls]: Yes, make something closer to the `node-alerts-rules` one please (you shouldn't take the same name btw otherwise you will overwrite the other one)

File: salt/metalk8s/addons/prometheus-operator/deployed/kube-alerts-rules.sls [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+{%- set rules = prometheus.get('spec', {}).get('rules', {}) %}
+
+{%- raw %}
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  labels:
+    app.kubernetes.io/part-of: metalk8s
+    metalk8s.scality.com/monitor: ''
+    release: prometheus-operator
```
Comment (TeddyAndrieux) [inline salt/metalk8s/addons/prometheus-operator/deployed/kube-alerts-rules.sls]: Remove this one please

File: salt/metalk8s/addons/prometheus-operator/deployed/kube-alerts-rules.sls [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    - alert: KubeJobNotCompleted
+      annotations:
+        description: Job {{ $labels.namespace }}/{{ $labels.job_name }} is taking
+          more than {% endraw %} {{ rules.kube_apps.kube_job_not_completed.warning.hours }} {% raw %} hours to complete.
+        runbook_url: https://runbooks.prometheus-operator.dev/runbooks/kubernetes/kubejobnotcompleted
+        summary: Job did not complete in time
+      expr: |-
+        time() - max by(namespace, job_name, cluster) (kube_job_status_start_time{job="kube-state-metrics", namespace=~".*"}
+          and
+        kube_job_status_active{job="kube-state-metrics", namespace=~".*"} > 0) > {% endraw %} {{ rules.kube_apps.kube_job_not_completed.warning.hours }} {% raw %}*60*60
```
Comment (TeddyAndrieux) [inline salt/metalk8s/addons/prometheus-operator/deployed/kube-alerts-rules.sls]: Here you will have extra spaces, also I would add parenthesis at the end just to be sure

```suggestion
        kube_job_status_active{job="kube-state-metrics", namespace=~".*"} > 0) > ({% endraw %}{{ rules.kube_apps.kube_job_not_completed.warning.hours }}{% raw %} * 60 * 60)
```

**Review by TeddyAndrieux — COMMENTED** [review]
Just two last minor comments and good to me (if you may also squash a bit the commit to remove the "fixes" commit)

File: salt/metalk8s/addons/prometheus-operator/deployed/kube-alerts-rules.sls [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+  name: metalk8s-kube-apps.rules
+  namespace: metalk8s-monitoring
+spec:
+  groups:
+  - name: kubernetes-apps
+    rules:
+    - alert: KubeJobNotCompleted
+      annotations:
+        description: Job {{ $labels.namespace }}/{{ $labels.job_name }} is taking
+          more than {% endraw %} {{ rules.kube_apps.kube_job_not_completed.warning.hours }} {% raw %} hours to complete.
```
Comment (TeddyAndrieux) [inline salt/metalk8s/addons/prometheus-operator/deployed/kube-alerts-rules.sls]: ditto btw

```suggestion
          more than {% endraw %}{{ rules.kube_apps.kube_job_not_completed.warning.hours }}{% raw %} hours to complete.
```

**Review by TeddyAndrieux — APPROVED** [review]

**Comment by eg-ayoub** [comment]
ping

**Comment by eg-ayoub** [comment]
ping
